packages/fluentui/react-northstar/src/components/Tree/hooks/flattenTree.ts

```diff
@@ -117,33 +118,33 @@ function flattenSubTree(
       visibleItemIds.push(id);
     }
 
-    if (hasSubtree) {
-      const { selectedChildrenNum, selectableChildrenNum } = flattenSubTree(
-        childrenItems as ObjectShorthandValue<TreeItemProps>[],
-        level + 1,
-        id,
-        flatTree,
-        isParentVisible && expanded, // parent being visible and expanded means subtree is visible
-        activeItemIds,
-        visibleItemIds,
-        selectedItemIds,
-      );
-
-      if (selectedChildrenNum === selectableChildrenNum) {
+    const { selectedChildrenNum, selectableChildrenNum } = flattenSubTree(
+      childrenItems as ObjectShorthandValue<TreeItemProps>[],
+      level + 1,
+      id,
+      flatTree,
+      isParentVisible && expanded, // parent being visible and expanded means subtree is visible
+      activeItemIds,
+      visibleItemIds,
+      selectedItemIds,
+    );
+
+    if (selectable) {
+      selectableNum++;
+
+      if (hasSubtree) {
+        if (selectedChildrenNum === selectableChildrenNum) {
+          flatTree[id].selected = true;
+          selectedNum++;
+        } else if (selectedChildrenNum > 0) {
+          flatTree[id].selected = 'indeterminate';
+          selectedNum += 0.5; // trick to propagate indeterminate state to ancestors
+        }
+      } else if (selectedItemIds.indexOf(id) >= 0) {
+        // selectedItemIds only make sense for leaf nodes
         flatTree[id].selected = true;
         selectedNum++;
```

Comment:
```suggestion
    if (selectable) {
      selectableNum++;
      if (
        (selectedChildrenNum === selectableChildrenNum && hasSubtree)
        || selectedItemIds.indexOf(id) >= 0
      ) {
        flatTree[id].selected = true;
        selectedNum++;
      } else if (selectedChildrenNum > 0) {
        flatTree[id].selected = 'indeterminate';
        selectedNum += 0.5; // trick to propagate indeterminate state to ancestors
      }
    }
```

Reply:
Avoids duplication for selecting a node

Reply:
Perhaps this might also be a nice addition

```javascript
(selectedChildrenNum === selectableChildrenNum && hasSubtree) // children are all selected
|| selectedItemIds.indexOf(id) >= 0 // leaf node
```

Reply:
```suggestion
   if (selectable) {
      selectableNum++;
      if (
        (hasSubtree && selectedChildrenNum === selectableChildrenNum)
        || (!hasSubtree && selectedItemIds.indexOf(id) >= 0) // selectedItemIds only make sense for leaf nodes
      ) {
        flatTree[id].selected = true;
        selectedNum++;
      } else if (selectedChildrenNum > 0) {
        flatTree[id].selected = 'indeterminate';
        selectedNum += 0.5; // trick to propagate indeterminate state to ancestors
      }
    }
```